### PR TITLE
[bitnami/thanos] feat: update default prometheusrule value

### PR DESCRIPTION
**Pull request by aslafy-z**
### Description of the change

Move from `metrics.prometheusRule.rules` to `metrics.prometheusRule.groups` and change the default value. 

### Benefits

Allow install with default PrometheusRule enabled without errors.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/thanos/values.yaml [hunk 1]
```diff
@@ -3838,7 +3838,7 @@ metrics:
     ##              for: 5m
     ##              labels:
     ##                severity: warning
-    rules: []
+    rules: {}
```
Comment (joancafom) [inline bitnami/thanos/values.yaml:3841]: Good point! But I think the real issue here is that `metrics.prometheusRule.rules` should be renamed to `metrics.prometheusRule.groups`, which is what they really define:

https://github.com/bitnami/charts/blob/41a76aa5520025ffb5dfa5bbc7ddbdb1627390f9/bitnami/thanos/templates/prometheusrule.yaml#L17-L19

According to documentation, `groups` is an array, so I would keep its value set to `[]`:

https://docs.okd.io/4.10/rest_api/monitoring_apis/prometheusrule-monitoring-coreos-com-v1.html#spec-groups

In order to address this issue, you may need to perform the following 3 actions:

1. Rename `metrics.prometheusRule.rules` to `metrics.prometheusRule.groups`, and change every occurrence in the chart to match the new name

2. Update the example in values.yaml: 

```yaml
    ## @param metrics.prometheusRule.groups  Content of Prometheus Rule file for Thanos
    ## These are just examples rules, please adapt them to your needs.
    ##    groups:
    ##      - name: Compactor
    ##        rules:
    ##          - alert: ThanosCompactMultipleRunning
    ##            annotations:
    ##              description: No more than one Thanos Compact instance should be running at once. There are {{`{{`}}$value{{`}}`}} instances running.
    ##              runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanoscompactmultiplerunning
    ##              summary: Thanos Compact has multiple instances running.
    ##            expr: sum by (job) (up{job=~"{{ template "common.names.fullname" . }}-compact.*"}) > 1
    ##            for: 5m
    ##            labels:
    ##              severity: warning
```

3. Slightly adapt the contents of `templates/prometheusrule.yaml` file:

```diff
{{- if and .Values.metrics.enabled .Values.metrics.prometheusRule.enabled }}
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  name: {{ template "common.names.fullname" . }}
  namespace: {{ default .Release.Namespace .Values.metrics.prometheusRule.namespace | quote }}
  labels: {{- include "common.labels.standard" . | nindent 4 }}
    {{- if .Values.metrics.prometheusRule.additionalLabels }}
    {{- include "common.tplvalues.render" (dict "value" .Values.metrics.prometheusRule.additionalLabels "context" $) | nindent 4 }}
    {{- end }}
    {{- if .Values.commonLabels }}
    {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 4 }}
    {{- end }}
  {{- if .Values.commonAnnotations }}
  annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 4 }}
  {{- end }}
spec:
+  groups: {{- include "common.tplvalues.render" ( dict "value" .Values.metrics.prometheusRule.groups "context" $ ) | nindent 2 }}
{{- end }}
```

Could you implement the changes? If this is too much of a hassle, I can craft a PR myself. Just let me know 😁

**Review by joancafom — CHANGES_REQUESTED** [review]
Thanks for the PR @aslafy-z ! I left you a comment, could you please take a look at it?

File: bitnami/thanos/values.yaml (hunk 1 above)
Comment (aslafy-z) [reply bitnami/thanos/values.yaml:3841]: I justed tried and it fails with
```
Error: YAML parse error on thanos/templates/prometheusrule.yaml: error converting YAML to JSON: yaml: line 14: could not find expected ':'
helm.go:84: [debug] error converting YAML to JSON: yaml: line 14: could not find expected ':'
```

I guess this is due to the "tplvalues" logic that tries to make helm compute [] as a template string. It works with objects however.

Comment (aslafy-z) [reply bitnami/thanos/values.yaml:3841]: I just updated the proposal. Please have a new check :)

**Comment by aslafy-z** [comment]
I bumped the minor version since this PR includes a breaking change.

File: bitnami/thanos/values.yaml (hunk 1 above)
Comment (joancafom) [reply bitnami/thanos/values.yaml]: Thanks! Just added a minor remark :)

**Comment by joancafom** [comment]
Apart from that, you will have to merge changes in master to your branch and solve the conflicts in `bitnami/thanos/Chart.yaml`

File: bitnami/thanos/values.yaml [hunk 2]
```diff
@@ -3823,22 +3823,21 @@ metrics:
...
+    ##         - alert: ThanosCompactMultipleRunning
+    ##           annotations:
+    ##             description: No more than one Thanos Compact instance should be running at once. There are {{`{{`}}$value{{`}}`}} instances running.
+    ##             runbook_url: https://github.com/thanos-io/thanos/tree/main/mixin/runbook.md#alert-name-thanoscompactmultiplerunning
+    ##             summary: Thanos Compact has multiple instances running.
+    ##           expr: sum by (job) (up{job=~"{{ template "common.names.fullname" . }}-compact.*"}) > 1
+    ##           for: 5m
+    ##           labels:
+    ##             severity: warning
+    groups: ""
```
[the remark this answers is not in the document]
Comment (aslafy-z) [reply bitnami/thanos/values.yaml]: Right, I just applied the change :)

**Review by joancafom — APPROVED** [review]
LGTM, thanks for your changes 🚀

**Pull request merged**